### PR TITLE
feat(parse): add opt-in default-subcommand flag routing

### LINKED ISSUE #1411: default_subcommand: accept the default command's flags in prefix position
`default_subcommand` only fires on a word (`ex firefox` → the default). An unknown flag at the parent errors first, so `em -ua @world` is `UnknownFlag -u` even though `-u`/`-a` live on the default command.

Would you take an opt-in (e.g. `default_subcommand_flags`) that peeks the first selector word, skipping flags declared on the current command *or* the default, then:

- sibling name (`query`, `search`) → parse as today
- only parent-known flags (`--help`, `-p`) → stay on the parent
- otherwise (`firefox`, `-ua @world`, `-r`) → descend into the default *before* binding tokens

That would let a CLI drop an argv-rewriter. Needs to stay opt-in so `mise --only-on-run build` does not change.

### PR DESCRIPTION
**Pull request by jdx**
`em -ua @world` currently fails at the parent even when `-u` and `-a` belong to its default command. Add `default_subcommand_flags #true` (or `#[usage(default_subcommand = "install", default_subcommand_flags)]`) so it parses as `em install -ua @world`.

Lookahead skips recognized parent/default flags and their values. Explicit command names and aliases retain precedence, parent-only flags and bare invocations stay on the parent, and unknown flags stop lookahead. The implicit boundary goes before the first default-only flag, preserving parent flags before it and in the same short bundle (in either order), with normal child/global scope on subsequent tokens. Clause separators remain parser syntax during lookahead. Existing routing and legacy short-bundle behavior remain unchanged without the opt-in.

Includes Rust and Go runtime support, KDL/derive/generator plumbing, completion support, and documentation. Closes #1411.

Validation:
- Latest head `947d438b`: Linux CI, Windows CI, and Rust 1.91 MSRV checks pass, including workspace Rust/Go tests, minimal-feature checks, lint, and rendering.
- 43 shared argv cases and 4 completion cases, plus 7 derive/roundtrip/conflict tests. Review regressions cover mixed short bundles, parent ownership, clause separators, conflict diagnostics, and built-in shorts.
- Local Rust argv/library suites, conformance argv/reference/derive/completion suites, full Go suite, all-feature Clippy, and the full CI minimal-feature sequence pass.
- The earlier local zsh PTY timeout also reproduces on unchanged base `6a037426`; the Linux CI shell tests pass.
- Performance comparison is queued behind other repositories on the shared dispatcher; no measurement is available yet.

---

> [!NOTE]
> **High Risk**
> Changes core argv parsing and completion across Rust, Go, and lib paths with subtle boundary and short-bundle rules; mistakes would mis-bind flags or break existing default-subcommand CLIs that opt in.
> 
> **Overview**
> Adds **`default_subcommand_flags`** so CLIs can treat leading flags as belonging to the configured default subcommand (e.g. `em -ua @world` → `em install -ua @world`) without typing the subcommand name.
> 
> When enabled, parsers run a **read-only lookahead** to place an implicit command boundary before the first default-only flag, then bind as usual. **Parent and explicit sibling commands keep precedence**; parent-only flags and mixed short bundles (`-pu` / `-up`) still bind to the parent on the boundary token; unknown flags and `--` stop lookahead. Behavior is **opt-in** everywhere (KDL, `#[usage(...)]`, Rust `usage-argv`, `usage-lib` parse, Go `argv`, completion, lint, codegen).
> 
> Ships reference docs, a large conformance corpus, derive round-trip/ conflict tests, and validation that **`default_subcommand_flags` requires `default_subcommand`**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 947d438b0e73f8403359f8119b080ea6470603a3. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added opt-in support for parsing default subcommand flags before the first positional argument.
  * Preserved parent-flag behavior in mixed short-flag bundles and supported attached or detached values.
  * Extended argument completion to suggest flags from the configured default subcommand.
  * Added support across Rust and Go command configurations and generated specifications.
* **Bug Fixes**
  * Added validation when routing is enabled without a default subcommand.
* **Documentation**
  * Documented routing, boundary behavior, and configuration options.
* **Tests**
  * Added coverage for parsing, completion, round-trips, conflicts, separators, and edge cases.

### CODE REVIEW

File: argv/src/lib.rs [hunk 1]
```diff
@@ -1881,6 +2021,18 @@ impl<'t: 'v, 'a, 'v> Parser<'t, 'a, 'v> {
...
+            self.default_flag_at = None;
+            if let Some(default) = self.cmd.default_subcommand {
+                if self.cmd.args_conflicts_with_subcommands && self.command_arg_found {
+                    return Some(Err(Error::SubcommandConflict {
+                        subcommand: default,
+                    }));
+                }
+                self.default_taken = true;
+                return Some(self.descend(default).map(|()| Event::Command(default)));
+            }
```
[the remark this answers is not in the document]
Comment (jdx) [reply argv/src/lib.rs:2076]: Fixed in 159f8fea. The boundary short bundle now retains parent flag declarations and ownership in Rust, Go, and the reference parser, including `-pua`, `-upa`, and `-uap`. The extra parent scope ends with that token. Added shared corpus cases for both orders, parent values, shadowed shorts, and scope isolation, plus a derive binding test. Parent/subcommand conflicts also apply within the bundle.

_This comment was generated by Codex._

File: lib/src/parse.rs [hunk 2]
```diff
@@ -1476,7 +1594,14 @@ fn parse_partial_traced(
...
         }
-        if let Some(subcommand) = out.cmd.find_subcommand(&input[idx].word) {
+        let implicit_default = !used_default_subcommand && default_flag_at == Some(idx);
+        let selected_command = if implicit_default {
+            out.cmd
+                .find_subcommand(spec.default_subcommand.as_deref().expect("default route"))
+        } else {
+            out.cmd.find_subcommand(&input[idx].word)
+        };
+        if let Some(subcommand) = selected_command {
```
[the remark this answers is not in the document]
Comment (jdx) [reply lib/src/parse.rs:1641]: Fixed in 159f8fea. The conflict message now uses the selected subcommand name, so implicit routing reports `install` rather than `-u`. Added a reference diagnostic regression and mixed-bundle conflict checks for Rust and Go.

_This comment was generated by Codex._

File: argv/src/lib.rs [hunk 3]
```diff
@@ -1756,11 +1765,142 @@ impl<'t: 'v, 'a, 'v> Parser<'t, 'a, 'v> {
...
+                        .map(bytes)
+                        .filter(|next| {
+                            flag.allow_hyphen_values
+                                || !is_flag_like(next)
+                                || (flag.allow_negative_numbers && is_negative_number(next))
+                        })
+                        .inspect(|_| i += 1)
+                } else {
+                    None
+                };
```
[the remark this answers is not in the document]
Comment (jdx) [reply argv/src/lib.rs:1882]: Fixed in 159f8fea across Rust runtime, reference parser, and Go. Lookahead stops at the active command's clause separator, excludes it from detached flag values, and stops variadic collection before it. The active scope is the parent before implicit routing and the default command afterward. Shared corpus regressions cover optional/required root flag values and a default-command separator followed by a sibling name.

_This comment was generated by Codex._

File: go/argv/parser.go [hunk 4]
```diff
@@ -95,11 +102,132 @@ type Parser struct {
...
+							flag = f
+							break
+						}
+					}
+				}
+				if flag == nil {
+					return at
+				}
+				if parent == nil && at < 0 {
+					at = i
```
[the remark this answers is not in the document]
Comment (jdx) [reply go/argv/parser.go:189]: Fixed in 159f8fea in Go and both Rust parsers. A mixed boundary bundle preserves parent ownership in either order; following tokens use normal child/global scope. Shared corpus cases cover parent-first/child-first bundles, attached and detached parent values, shadowed shorts, unknown letters, and scope isolation. Full `go test ./...` passes.

_This comment was generated by Codex._

File: lib/src/parse.rs [hunk 5]
```diff
@@ -1476,13 +1631,45 @@ fn parse_partial_traced(
...
+                    for short in input[idx].word[1..].chars() {
+                        let key = format!("-{short}");
+                        if out.available_flags.contains_key(&key) {
+                            found = true;
+                            break;
+                        }
+                        match child_flags.get(&key) {
+                            Some(flag) if flag.arg.is_none() => {}
+                            _ => break,
+                        }
```
[the remark this answers is not in the document]
Comment (jdx) [reply lib/src/parse.rs:1660]: Fixed in 947d438b. The reference conflict scan now calls the existing `supplied_short` lookup, matching Rust and Go for enabled built-in flags. Added `-uhp` and `-uVp` regressions to the reference diagnostic, derived runtime, and Go tests; all pass.

_This comment was generated by Codex._

**Comment by jdx** [comment]
@coderabbitai review

Please review the latest head, 947d438b. The prior review status still points at 159f8fea; the mixed-bundle, clause-separator, and built-in conflict fixes are now pushed and tested.

_This comment was generated by Codex._
